### PR TITLE
overlord/boot: have firstboot support assertion files with multiple assertions

### PR DESCRIPTION
**Pull request by pedronis**
For that case also  be prepared for repetitions.

### CODE REVIEW

File: overlord/boot/firstboot.go [hunk 1]
```diff
@@ -137,6 +139,27 @@ func populateStateFromSeed() error {
 	return ovld.Stop()
 }
 
+func readAsserts(fn string) ([]asserts.Assertion, error) {
+	res := ([]asserts.Assertion)(nil)
```
Comment (niemeyer) [inline overlord/boot/firstboot.go]: The usual spelling here is
```
var res []asserts.Assertion
```
Might even just name the return variables (`(res ..., err error)`).

File: overlord/boot/firstboot.go [hunk 2]
```diff
@@ -154,26 +177,31 @@ func importAssertionsFromSeed(st *state.State) error {
...
+						// we already got something more recent
+						continue
+					}
+				}
+				return err
+			}
+			assertionsToAdd = append(assertionsToAdd, as)
+			if as.Type() == asserts.ModelType {
+				if modelAssertion != nil {
+					return fmt.Errorf("cannot add more than one model assertion")
```
Comment (niemeyer) [inline overlord/boot/firstboot.go]: Nice that the logic ought to work even if we have two model assertions, as long as they have matching keys, due to the revision logic above.

**Comment by niemeyer** [comment]
Super trivial. LGTM regardless.

File: overlord/boot/firstboot_test.go [hunk 3]
```diff
@@ -229,6 +229,143 @@ snaps:
...
+		"publisher-id": "developerid",
+		"snap-name":    "foo",
+		"timestamp":    time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	sha3_384, size, err := asserts.SnapFileSHA3_384(fooSnapFile)
+	c.Assert(err, IsNil)
+
+	fmt.Println("foo", sha3_384)
```
Comment (mvo5) [inline overlord/boot/firstboot_test.go]: debug print?

File: overlord/boot/firstboot_test.go [hunk 4]
```diff
@@ -229,6 +229,143 @@ snaps:
...
+		"publisher-id": "developerid",
+		"snap-name":    "bar",
+		"timestamp":    time.Now().UTC().Format(time.RFC3339),
+	}, nil, "")
+	c.Assert(err, IsNil)
+
+	sha3_384, size, err = asserts.SnapFileSHA3_384(barSnapFile)
+	c.Assert(err, IsNil)
+
+	fmt.Println("bar", sha3_384)
```
Comment (mvo5) [inline overlord/boot/firstboot_test.go]: another debug print?

**Comment by mvo5** [comment]
Looks very nice, thank you! Just two trivials.

**Pull request merged**